app/services/scraper/bedelias.rb

```diff
@@ -81,9 +83,25 @@ def go_to_prerequisites_page
   end
 
   def process_prerequisites(subjects)
+    Thread.abort_on_exception = true
+
     total_pages = /\(\d+ de (\d+)\)/.match(find(".ui-paginator-current").text).captures[0]
+    max_pages = MAX_PAGES || total_pages.to_i
+
+    1.upto(max_pages).each_slice((max_pages / THREADS).ceil).map do |slice|
+      Thread.new do
+        using_session(Capybara::Session.new(page.mode)) do
+          process_prerequisites_slice(subjects, slice, total_pages)
+        end
+      end
+    end.flat_map(&:value)
+  end
+
+  def process_prerequisites_slice(subjects, slice, total_pages)
+    go_to_groups_and_subjects_page
+    go_to_prerequisites_page
 
-    1.upto(MAX_PAGES || total_pages.to_i).flat_map do |page|
+    slice.flat_map do |page|
       Rails.logger.info "Page #{page}"
```

Comment:
The output looks a little bit weird now tho:

```
1640 - OPTIMIZACION CONTINUA Y APLICACIONES (Curso)
1224 - ECONOMIA (Examen)
1537 - SISTEMAS OPERATIVOS (Curso)
1736 - INT. A LA INTERAC. PERSONA COMPUTADORA (Curso)
Page 2
1743 - ARQUITECTURA DE SOFTWARE (Curso)
Page 5
1610 - INT. A LA INVESTIGACION DE OPERACIONES (Curso)
Page 11
1028 - ECUACIONES DIFERENCIALES (Curso)
1930 - ORGANIZACIONES PARA INGENIEROS (Examen)
1444 - ASPECTOS AVANZ. DE ARQ.DE COMPUTADORAS (Curso)
```

I wonder if it would be better foor now to remove the printing of the page number. We can see how we can add it again later if you want 🙂

Reply:
makes sense. Removed in 1813a85